server/eclipse-project/src/main/java/us/freeandfair/corla/util/NaturalOrderComparator.java

```diff
@@ -93,14 +93,22 @@ public int compare(final String an_a, final String an_b) {
     final Iterator<String> as = split(an_a).iterator();
     final Iterator<String> bs = split(an_b).iterator();
 
-    while (as.hasNext() && bs.hasNext()) {
+    while (as.hasNext() && bs.hasNext() && result == 0) {
       final String x = as.next();
       final String y = bs.next();
 
       if (isDigit(x) && isDigit(y)) {
         result = compareAsDigits(x, y);
       } else {
         result = String.CASE_INSENSITIVE_ORDER.compare(x, y);
```

Comment:
What if you passed `y, x` here with a similar comment you added below?

Reply:
that would work, but then we have values like -5 which is weird for compareTo

Reply:
OK, not a big deal to leave it the way you have it. However, the contract for `compare` is just to return a negative integer, a positive integer, or zero - not specifically `-1` or `1`, so if you hit this in the future, tests that check against those numbers specifically are brittle against the `Comparator` contract.